File: DailyChallenges/CPP/LeetCode_70.cpp
// 70. Climbing Stairs @ 2022/12/12
class Solution {
private:
    unordered_map<int, int> cache;
public:
    int climbStairs(int n) {
        if (cache[n] != 0) return cache[n];
        if ( n== 1 || n == 0) return (cache[n] = 1);
        return (cache[n] = climbStairs(n-1) + climbStairs(n-2));
    };
};

File: DailyChallenges/Scala/LeetCode_70.scala
object Solution {
    def climbStairs(n: Int): Int = {
        val steps = Array(1, 2)
        if (n <= 2) return steps(n-1)
        for (_ <- 3 to n){ val nxt = steps(0) + steps(1); steps(0) = steps(1); steps(1) = nxt}
        steps(1)
    }
}
